tests/calculations/test_pw.py

```diff
@@ -250,3 +250,67 @@ def test_pw_parallelization_duplicate_cmdline_flag(fixture_sandbox, generate_cal
     with pytest.raises(InputValidationError) as exc:
         generate_calc_job(fixture_sandbox, entry_point_name, inputs)
     assert 'Conflicting' in str(exc.value)
+
+
+@pytest.mark.parametrize('calculation', ['nscf', 'bands', 'scf', 'relax', 'vc-relax'])
+def test_pw_validate_inputs_restart(
+    fixture_sandbox, generate_calc_job, generate_inputs_pw, fixture_localhost, generate_remote_data, calculation
+):
+    """Test the input validation of restart settings for the ``PwCalculation``."""
+    remote_data = generate_remote_data(computer=fixture_localhost, remote_path='/cat_stevens')
+    entry_point_name = 'quantumespresso.pw'
+
+    inputs = generate_inputs_pw()
+    parameters = inputs['parameters'].get_dict()
+    parameters['CONTROL']['calculation'] = calculation
+
+    if calculation in ('nscf', 'bands'):
+
+        # No parent_folder -> raise
+        inputs['parameters'] = orm.Dict(dict=parameters)
+        with pytest.raises(ValueError) as exc:
+            generate_calc_job(fixture_sandbox, entry_point_name, inputs)
+        assert f'`parent_folder` not provided for `{calculation}` calculation.' in str(exc.value)
```

Comment:
If you want to check the exception message, the following is recommended
```suggestion
        with pytest.raises(ValueError, match=r'`parent_folder` not provided for `.*` calculation.'):
            generate_calc_job(fixture_sandbox, entry_point_name, inputs)
```